### PR TITLE
Migrate kill tests to ginkgo

### PR DESCRIPTION
**Pull request by baude**
Migrate kill tests to the ginkgo suite and remove the
podman_kill bats.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: test/e2e/kill_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	It("podman kill a running container by id", func() {
+		session := podmanTest.RunSleepContainer("")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		cid := session.OutputToString()
+
+		result := podmanTest.Podman([]string{"kill", cid})
+		result.WaitWithDefaultTimeout()
+		Expect(result.ExitCode()).To(Equal(0))
+	})
```
Comment (rhatdan) [inline test/e2e/kill_test.go]: Could you add a check to make sure the container is no longer running.
```
+		result := podmanTest.Podman([]string{"stats", cid})
+		result.WaitWithDefaultTimeout()
 +		Expect(session.ExitCode()).To(Not(Equal(0))) 
```

File: test/e2e/kill_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	It("podman kill a running container by id with TERM", func() {
+		session := podmanTest.RunSleepContainer("")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		cid := session.OutputToString()
+
+		result := podmanTest.Podman([]string{"kill", "-s", "TERM", cid})
+		result.WaitWithDefaultTimeout()
+		Expect(result.ExitCode()).To(Equal(0))
+	})
```
Comment (rhatdan) [inline test/e2e/kill_test.go]: Ditto

File: test/e2e/kill_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+	It("podman kill a running container by name", func() {
+		session := podmanTest.RunSleepContainer("test1")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		result := podmanTest.Podman([]string{"kill", "-s", "TERM", "test1"})
+		result.WaitWithDefaultTimeout()
+		Expect(result.ExitCode()).To(Equal(0))
+	})
```
Comment (rhatdan) [inline test/e2e/kill_test.go]: ditto

File: test/e2e/kill_test.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+	It("podman kill latest container", func() {
+		session := podmanTest.RunSleepContainer("")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		result := podmanTest.Podman([]string{"kill", "-l"})
+		result.WaitWithDefaultTimeout()
+		Expect(result.ExitCode()).To(Equal(0))
+	})
```
Comment (rhatdan) [inline test/e2e/kill_test.go]: ditto

File: test/e2e/kill_test.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+		result := podmanTest.Podman([]string{"kill", "-s", "foobar", cid})
+		result.WaitWithDefaultTimeout()
+		Expect(result.ExitCode()).To(Equal(125))
+	})
+
+	It("podman kill latest container", func() {
+		session := podmanTest.RunSleepContainer("")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
```
Comment (rhatdan) [inline test/e2e/kill_test.go]: +		cid := session.OutputToString()

**Comment by mheon** [comment]
LGTM, though we definitely need to do a pass over our integration tests once they're all converted over and we can really take advantage of Ginkgo

**Comment by TomSweeneyRedHat** [comment]
LGTM, the more of the ginkgo tests I see, the better I like it!
Red Hat CI is unhappy without any clues as to what's wrong.  Retest?

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
bot, retest this

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**